packages/orchestration/src/service.js

```diff
@@ -104,22 +133,53 @@ const prepareOrchestration = (zone, makeChainAccountKit) =>
             chainAccountKit.connectionHandler,
           );
           // XXX if we fail, should we close the port (if it was created in this flow)?
-
           return chainAccountKit.account;
         },
+        /**
+         * @param {IBCConnectionID} controllerConnectionId
+         * @returns {Promise<ICQConnection>}
+         */
+        async provideICQConnection(controllerConnectionId) {
+          if (this.state.icqConnections.has(controllerConnectionId)) {
+            return this.state.icqConnections.get(controllerConnectionId)
+              .connection;
+          }
+          // allocate a new Port for every Connection
```

Comment:
I am not sure if this is the correct behavior. An alternative is to bind a single port during the `orchestration-proposal.js`. Happy to fix here, or create a small task to capture the work.

Reply:
seems worth its own issue; something like:

 - optimize ICQ port allocation

There I'd like us to find an ICQ spec that tells us what the right answer is, and/or do some integration testing.

Reply:
Created #9317 and included it as a code comment